packages/frontend/src/features/metricsCatalog/components/visualization/MetricsVisualization.tsx

```diff
@@ -444,6 +449,42 @@ const MetricsVisualization: FC<Props> = ({
         results?.metric.timeDimension?.field,
     ]);
 
+    const splitSegments = useMemo(() => {
+        return segmentedData.map((segment) => {
+            const { data: segmentData, ...rest } = segment;
+            const completedPeriodData: MetricExploreDataPointWithDateValue[] =
+                [];
+            const incompletePeriodData: MetricExploreDataPointWithDateValue[] =
+                [];
+
+            segmentData.forEach((row) => {
+                if (
+                    isInCurrentTimeFrame(
+                        row.date,
+                        results?.metric.timeDimension?.interval,
+                    )
+                ) {
+                    incompletePeriodData.push(row);
+                } else {
+                    completedPeriodData.push(row);
+                }
+            });
+
+            const lastCompletedPeriodData = completedPeriodData.sort(
+                (a, b) => b.date.getTime() - a.date.getTime(),
+            )[0];
+
+            // Add the last completed period data to the incomplete period data, this will fill in the gap between the last completed period and the first incomplete period
+            incompletePeriodData.push(lastCompletedPeriodData);
+
+            return {
+                ...rest,
+                completedPeriodData,
+                incompletePeriodData,
+            };
+        });
+    }, [results?.metric.timeDimension?.interval, segmentedData]);
```

Comment:
would be interesting to move this to the react-query hook as a `select` - but an improvement for later